_posts/2020-09-10-ompl-constrained-planning-gsoc.md

```diff
@@ -0,0 +1,52 @@
+
+The most interesting aspect of this new planning approach is shown below. Planning with end-effector constraints sometimes results in large joint space jumps, making the path unusable on a real robot (shown on the left). The new planner solves the same problem without these joint space jumps in the solution (shown on the right). For constraints such as equality position constraints, the existing way to avoid joint space jumps, using the [enforce_joint_model_state_space](https://ros-planning.github.io/moveit_tutorials/doc/ompl_interface/ompl_interface_tutorial.html#enforce-planning-in-joint-space) option, does not work.
+
+<img src="/assets/images/blog_posts/ompl_constrained_planning/pos_con_pose_model.gif" alt="drawing" width="45%"/>
+<img src="/assets/images/blog_posts/ompl_constrained_planning/pos_con_new_planner.gif" alt="drawing" width="45%"/>
+
+A [new tutorial](https://github.com/ros-planning/moveit_tutorials/pull/518) explains how to configure and use this new planning option, using simple but instructive examples with the Panda robot. These are implemented using MoveIt’s [Python interface](https://ros-planning.github.io/moveit_tutorials/doc/move_group_python_interface/move_group_python_interface_tutorial.html), making it extremely convenient to try out the new planner and quickly experiment with different planning problems.
+
+<img src="/assets/images/blog_posts/ompl_constrained_planning/tutorial_case_2.gif" alt="drawing" width="45%"/>
+<img src="/assets/images/blog_posts/ompl_constrained_planning/tutorial_case_3.gif" alt="drawing" width="45%"/>
+
+These example problems are only the tip of the iceberg. An [external planning plugin](https://github.com/JeroenDM/elion) showes the potential of this new planning technique and allows you to test experimental constraints. In addition, alternative approaches to projection-based sampling are available im OMPL that could be used for a subset of the constraints.
+
+<img src="/assets/images/blog_posts/ompl_constrained_planning/welding_example.gif" alt="drawing" width="45%"/>
+<img src="/assets/images/blog_posts/ompl_constrained_planning/kuka_pos_con.gif" alt="drawing" width="45%"/>
+
+The project also improved the exiting code in many ways. More than 100 lines of unused code are removed, the logging statements are cleaned up and a surprising [bug](https://github.com/ros-planning/moveit/pull/2239) is fixed. See [here](https://gist.github.com/JeroenDM/426e3a7e083049295bbcb660c9a98e63) for a complete list of the pull requests.
+
+
+## Limitations
+
+The main (temporary) drawback is that the code is still being review and merged. So it can only be used by building MoveIt from source right now. Hopefully, we can quickly work through the review process. Feel free to provide feedback and help out!
```

Comment:
```suggestion
The main (temporary) drawback is that the code is still being reviewed and merged. So it can only be used by building MoveIt from source right now. Hopefully, we can quickly work through the review process. Feel free to provide feedback and help out!
```